File: Level 1/Rotary Lock 1/javascript/Rotary_Lock_1.js
/**
 * @param {number} N
 * @param {number} M
 * @param {number[]} C
 * @return {number}
 */
function getMinCodeEntryTime(N, M, C) {
    time = 0
    prev = 1
    for (i = 0; i < M; i++) {
      digit = C[i]
      time += Math.min(Math.abs(digit - prev), N + Math.min(prev, digit) - Math.max(prev, digit))
      prev = digit
    }
    
    return time
  }

File: Level 1/Rotary Lock 1/python/Rotary_Lock_1.py
from typing import List

def getMinCodeEntryTime(N: int, M: int, C: List[int]) -> int:
    codeEntryTime = 0
    lockPos = 1
    for i in range(M):
        digit = C[i]

        # Get the time to rotate the lock to the next digit and update the lock position
        codeEntryTime += getRotationTime(N, lockPos, digit)
        lockPos = digit
    
    return codeEntryTime

def getRotationTime(N: int, startDigit: int, endDigit: int) -> int:
    return min(abs(endDigit - startDigit), N + min(startDigit, endDigit) - max(startDigit, endDigit))